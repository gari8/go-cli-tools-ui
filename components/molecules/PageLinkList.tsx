import PageLink from "@/components/atoms/PageLink";
import { Box } from "@chakra-ui/react";
import { IPageObject } from "@/types/ui";
import { ImPriceTags } from "react-icons/im";
import { IoMdCreate } from "react-icons/io";
import { BsGearWideConnected, BsTools } from "react-icons/bs";

interface Props {
  pageList?: IPageObject[];
  selectedColor?: string;
  pointerColor?: string;
}

export default function PageLinkList({
  pageList,
  selectedColor,
  pointerColor,
}: Props) {
  const defaultPageList = [
    {
      name: "Tools",
      url: "/",
      icon: BsTools,
    },
    {
      name: "Tags",
      url: "/tags",
      icon: ImPriceTags,
    },
    {
      name: "Register",
      url: "/register",
      icon: IoMdCreate,
    },
    {
      name: "Setting",
      url: "/setting",
      icon: BsGearWideConnected,
    },
  ];

  return (
    <Box>
      {(pageList ? pageList : defaultPageList).map(
        (page: IPageObject, index: number) => {
          return (
            <PageLink
              page={page}
              key={page.name + index}
              selectedColor={selectedColor}
              pointerColor={pointerColor}
            />
          );
        }
      )}
    </Box>
  );
}
