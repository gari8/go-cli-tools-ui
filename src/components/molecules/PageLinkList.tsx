import PageLink from "~/components/atoms/PageLink";
import { Box } from "@chakra-ui/react";
import { IPageObject } from "~/types/ui";
import { TiSpanner } from "react-icons/ti";

interface Props {
  pageList?: IPageObject[];
  selectedColor?: string;
  pointerColor?: string;
}

export default function PageLinkList({
  pageList,
  selectedColor,
  pointerColor,
}: Props) {
  const defaultPageList = [
    {
      name: "Tools",
      url: "/",
      icon: TiSpanner,
    },
    {
      name: "Tags",
      url: "/tags",
    },
    {
      name: "Register",
      url: "/register",
    },
    {
      name: "Setting",
      url: "/setting",
    },
  ];

  return (
    <Box>
      {(pageList ? pageList : defaultPageList).map(
        (page: IPageObject, index: number) => {
          return (
            <PageLink
              page={page}
              key={page.name + index}
              selectedColor={selectedColor}
              pointerColor={pointerColor}
            />
          );
        }
      )}
    </Box>
  );
}
