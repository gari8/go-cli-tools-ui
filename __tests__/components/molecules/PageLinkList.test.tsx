import { render } from "@testing-library/react";
import PageLinkList from "@/components/molecules/PageLinkList";

describe("PageLinkList", () => {
  test("is components", () => {
    const { asFragment } = render(<PageLinkList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
