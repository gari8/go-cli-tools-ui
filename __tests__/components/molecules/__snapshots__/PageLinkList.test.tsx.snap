// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLinkList is components 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <div
      class="css-5tigzz"
    >
      <div
        class="css-1qz17lo"
      >
        <div
          class="css-lyvtfg"
        >
          <svg
            class="chakra-icon css-13otjrl"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M0 1l1-1 3.081 2.2a1 1 0 01.419.815v.07a1 1 0 00.293.708L10.5 9.5l.914-.305a1 1 0 011.023.242l3.356 3.356a1 1 0 010 1.414l-1.586 1.586a1 1 0 01-1.414 0l-3.356-3.356a1 1 0 01-.242-1.023L9.5 10.5 3.793 4.793a1 1 0 00-.707-.293h-.071a1 1 0 01-.814-.419L0 1zm11.354 9.646a.5.5 0 00-.708.708l3 3a.5.5 0 00.708-.708l-3-3z"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M15.898 2.223a3.003 3.003 0 01-3.679 3.674L5.878 12.15a3 3 0 11-2.027-2.027l6.252-6.341A3 3 0 0113.778.1l-2.142 2.142L12 4l1.757.364 2.141-2.141zm-13.37 9.019L3.001 11l.471.242.529.026.287.445.445.287.026.529L5 13l-.242.471-.026.529-.445.287-.287.445-.529.026L3 15l-.471-.242L2 14.732l-.287-.445L1.268 14l-.026-.529L1 13l.242-.471.026-.529.445-.287.287-.445.529-.026z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <p
          class="chakra-text css-0"
        >
          Tools
        </p>
      </div>
    </div>
    <div
      class="css-5tigzz"
    >
      <div
        class="css-cezcce"
      >
        <div
          class="css-lyvtfg"
        >
          <svg
            class="chakra-icon css-13otjrl"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 20 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.25 0h-6c-0.412 0-0.989 0.239-1.28 0.53l-7.439 7.439c-0.292 0.292-0.292 0.769 0 1.061l6.439 6.439c0.292 0.292 0.769 0.292 1.061 0l7.439-7.439c0.292-0.292 0.53-0.868 0.53-1.28v-6c0-0.412-0.337-0.75-0.75-0.75zM15.5 6c-0.828 0-1.5-0.672-1.5-1.5s0.672-1.5 1.5-1.5 1.5 0.672 1.5 1.5-0.672 1.5-1.5 1.5z"
            />
            <path
              d="M2 8.5l8.5-8.5h-1.25c-0.412 0-0.989 0.239-1.28 0.53l-7.439 7.439c-0.292 0.292-0.292 0.769 0 1.061l6.439 6.439c0.292 0.292 0.769 0.292 1.061 0l0.47-0.47-6.5-6.5z"
            />
          </svg>
        </div>
        <p
          class="chakra-text css-0"
        >
          Tags
        </p>
      </div>
    </div>
    <div
      class="css-5tigzz"
    >
      <div
        class="css-cezcce"
      >
        <div
          class="css-lyvtfg"
        >
          <svg
            class="chakra-icon css-13otjrl"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 368v80h80l235.727-235.729-79.999-79.998L64 368zm377.602-217.602c8.531-8.531 8.531-21.334 0-29.865l-50.135-50.135c-8.531-8.531-21.334-8.531-29.865 0l-39.468 39.469 79.999 79.998 39.469-39.467z"
            />
          </svg>
        </div>
        <p
          class="chakra-text css-0"
        >
          Register
        </p>
      </div>
    </div>
    <div
      class="css-5tigzz"
    >
      <div
        class="css-cezcce"
      >
        <div
          class="css-lyvtfg"
        >
          <svg
            class="chakra-icon css-13otjrl"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9.928 1.723c-.243-.97-1.62-.97-1.863 0l-.072.286a.96.96 0 01-1.622.435l-.204-.212c-.695-.718-1.889-.03-1.614.932l.08.283a.96.96 0 01-1.186 1.187l-.283-.081c-.961-.275-1.65.919-.932 1.614l.212.204a.96.96 0 01-.435 1.622l-.286.072c-.97.242-.97 1.62 0 1.863l.286.071a.96.96 0 01.435 1.622l-.212.205c-.718.695-.03 1.888.932 1.613l.283-.08a.96.96 0 011.187 1.187l-.081.283c-.275.96.919 1.65 1.614.931l.204-.211a.96.96 0 011.622.434l.072.286c.242.97 1.62.97 1.863 0l.071-.286a.96.96 0 011.622-.434l.205.212c.695.718 1.888.029 1.613-.932l-.08-.283a.96.96 0 011.187-1.188l.283.081c.96.275 1.65-.918.931-1.613l-.211-.205A.96.96 0 0115.983 10l.286-.071c.97-.243.97-1.62 0-1.863l-.286-.072a.96.96 0 01-.434-1.622l.212-.204c.718-.695.029-1.889-.932-1.614l-.283.08a.96.96 0 01-1.188-1.186l.081-.283c.275-.961-.918-1.65-1.613-.932l-.205.212A.96.96 0 0110 2.009l-.071-.286zm-.932 12.27a4.998 4.998 0 100-9.994 4.998 4.998 0 000 9.995z"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M8.372 8.996L5.598 5.298l.8-.6 2.848 3.798h4.748v1H9.246l-2.849 3.798-.8-.6 2.775-3.698z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <p
          class="chakra-text css-0"
        >
          Setting
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
